internal/parser/parser.go

```diff
@@ -2037,7 +2037,13 @@ func (p *Parser) parseModuleOrNamespaceDeclaration(pos int, hasJSDoc bool, modif
 	}
 	var body *ast.Node
 	if p.parseOptional(ast.KindDotToken) {
-		body = p.parseModuleOrNamespaceDeclaration(p.nodePos(), false /*hasJSDoc*/, nil /*modifiers*/, ast.NodeFlagsNestedNamespace|namespaceFlag)
+		syntheticExport := p.factory.NewModifier(ast.KindExportKeyword)
+		syntheticExport.Loc = core.NewTextRange(p.nodePos(), p.nodePos())
+		syntheticExport.Flags = ast.NodeFlagsSynthesized
+		nodes := p.nodeSlicePool.NewSlice(1)
+		nodes[0] = syntheticExport
+		syntheticModifiers := p.newModifierList(syntheticExport.Loc, nodes)
+		body = p.parseModuleOrNamespaceDeclaration(p.nodePos(), false /*hasJSDoc*/, syntheticModifiers, ast.NodeFlagsNestedNamespace|namespaceFlag)
```

Comment:
If we add a synthetic `export` keyword, can we do away with `NodeFlagsNestedNamespace`?

Reply:
It's even easier -- just check whether a namespace's parent is also a namespace (and that the parent's body is the namespace).